doc/source/dev/api-dev/array_api.rst

```diff
@@ -0,0 +1,253 @@
+Support for Array API
+=====================
+
+.. note:: Array API support is still experimental and hidden behind an
+          environment variable. Only a small part of the public API is covered
+          right now.
+
+This guide describes how to **use** and **add support for** the
+`Python Array API standard <https://data-apis.org/array-api/latest/index.html>`_.
+This standard allows users to use any Array API compatible array library
+without having to do anything.
+
+The `RFC`_ defines how SciPy implements support for the standard, with the main
+principle being *"array type in equals array type out"*. In addition, the
+implementation does more strict validation of allowed array-like inputs, e.g.
+rejecting numpy matrix and masked array instances, and arrays with object
+dtype.
+
+In the following, an Array API compatible namespace is noted as ``xp``.
+
+
+Using Array API support
+-----------------------
+
+To enable the Array API standard support, an environment variable must be set
+before importing SciPy:
+
+.. code:: bash
+
+   export SCIPY_ARRAY_API=1
```

Comment:
What‘s the reason to use an env variable? Why not global config option? A short comment would be nice.

Reply:
We have both. It's the same behaviour as in Scikit-Learn.

Reply:
Not quite - scikit-learn provides a context manager. And our global config is explicitly private; we're aiming for it to go away as soon as reasonably possible, it's only a tool to unblock development. Scikit-learn has public global config.

So the basic reason is that scikit-learn already had it, SciPy doesn't (and we'd prefer to keep it that way).

Reply:
I am adding this note in the docs here to address the review comment:

*Note that this environment variable is meant to be temporary, as a way to make incremental changes and merge them into ``main`` without affecting backwards compatibility immediately. We do not intend to keep this environment variable around long-term.*